lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -760,18 +756,62 @@ public static function get_style_variations( $scope = 'theme' ) {
 			$variation_files = array_merge( $variation_files, $variation_files_parent );
 		}
 		ksort( $variation_files );
+
+		return $variation_files;
+	}
+
+	/**
+	 * Returns the style variations defined by the theme (parent and child).
+	 *
+	 * @since 6.2.0 Returns parent theme variations if theme is a child.
+	 * @since 6.6.0 Added cache and configurable scope parameter to allow
+	 *              filtering theme.json partial files by the scope to
+	 *              which they can be applied e.g. theme vs block etc.
+	 *
+	 * @param string $scope The scope or type of style variation to retrieve e.g. theme, block etc.
+	 * @return array
+	 */
+	public static function get_style_variations( $scope = 'theme' ) {
+		$registry  = WP_Block_Type_Registry::get_instance();
+		$blocks    = $registry->get_all_registered();
+		$theme_dir = get_stylesheet_directory();
+		$locale    = get_locale();
+
+		if (
+			isset( static::$style_variations_cache[ $theme_dir ][ $locale ][ $scope ] ) &&
+			// Variations depend on registered blocks.
+			null !== static::$blocks &&
+			static::has_same_registered_blocks( 'theme' )
+		) {
+			return static::$style_variations_cache[ $theme_dir ][ $locale ][ $scope ];
+		}
+
+		$variation_files = static::get_style_variation_files();
+		$variations      = array(
+			'theme' => array(),
+			'block' => array(),
+		);
+
 		foreach ( $variation_files as $path => $file ) {
 			$decoded_file = wp_json_file_decode( $path, array( 'associative' => true ) );
-			if ( is_array( $decoded_file ) && static::style_variation_has_scope( $decoded_file, $scope ) ) {
-				$translated = static::translate( $decoded_file, wp_get_theme()->get( 'TextDomain' ) );
-				$variation  = ( new WP_Theme_JSON_Gutenberg( $translated ) )->get_raw_data();
-				if ( empty( $variation['title'] ) ) {
-					$variation['title'] = basename( $path, '.json' );
+			if ( is_array( $decoded_file ) ) {
+				$variation_scope = static::get_style_variation_scope( $decoded_file );
+
+				if ( $variation_scope ) {
+					$translated = static::translate( $decoded_file, wp_get_theme()->get( 'TextDomain' ) );
+					$variation  = ( new WP_Theme_JSON_Gutenberg( $translated ) )->get_raw_data();
```

Comment:
From what I'm seeing, many of Rust's block styles aren't registered and are therefore failing the test in `::get_blocks_metadata()`

https://github.com/WordPress/gutenberg/blob/d66f4d16bbe32bebeeff3be9d053a405e6b1cc9b/lib/class-wp-theme-json-gutenberg.php#L1080-L1084